src/main/java/seedu/agendum/model/ModelManager.java

```diff
@@ -82,6 +82,7 @@ public ReadOnlyToDoList getToDoList() {
 
     /** Raises an event to indicate the model has changed */
     private void indicateToDoListChanged() {
+        toDoList.resetData(toDoList);
```

Comment:
I was wondering if this had anything to do with our weird schedule bug for v0.3.
Maybe add some listener?
I considered binding completed task list to the main unfiltered list and hiding tasks which are not completed in the ui. But can't find java fx equivalent of display:none, visibility:hidden still take up space